content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+---
+reviewers:
+- robscott
+title: Topology Aware Hints
+feature:
+  title: Topology Aware Hints
+  description: >
+    Enables topology aware routing with topology hints on EndpointSlices.
+
+content_type: concept
+weight: 10
+---
+
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+_Topology Aware Hints_ enable topology aware routing with topology hints
+included in EndpointSlices. This approach tries to keep traffic close to where
+it originated from. This can result in decreased costs and improved performance.
+
+<!-- body -->
+
+## Introduction
+
+Kubernetes clusters are increasingly deployed in multi-zone environments.
+_Topology Aware Hints_ adds new functionality that can help keep traffic within
+the zone it originated from. This concept is commonly referred to as "Topology
+Aware Routing". The "Hints" here refer to a key implementation detail. The
+EndpointSlice controller adds hints to EndpointSlices to allocate each endpoint
+to a zone. Those hints are then consumed by components like kube-proxy as they
+configure how requests are routed.
+
+## Using Topology Aware Hints
+
+This feature can be enabled for individual Services by setting the
+`service.kubernetes.io/topology-aware-hints` annotation to `auto`. This tells
+the EndpointSlice controller to set topology hints if it is deemed safe.
+Importantly, this does not guarantee that hints will always be set.
+
+## How it Works
+
+The functionality enabling this feature is split into two components: The
+EndpointSlice controller and Kube-Proxy. This provides a high level overview of
+how each component implements this feature.
+
+### EndpointSlice controller
+
+The EndpointSlice controller is responsible for setting hints on EndpointSlices
+when this feature is enabled. The controller allocates a proportional amount of
+endpoints to each zone. This proportion is based on the number of CPU cores that
+have been allocated to nodes running in that zone. For example, if one zone had
+2 CPU cores and another zone only had 1 CPU core, the controller would allocated
+twice as many endpoints to the zone with 2 CPU cores.
```

Comment:
I would actually title this “Resource matching” or otherwise word it to emphasize that the hints are based on a specific resource, `cpu`.

Does this count:
- the amount of vCPU you get from summing all the nodes in that zone?
- the amount of vCPU you get from summing all the nodes in that zone, minus system reservations?
- the amount of vCPU you get from summing the `cpu` requests of Pods in that zone matching the selector for the relevant Service?
-  the amount of vCPU you get from summing the `cpu` limits of Pods in that zone matching the selector for the relevant Service?
- something else?

(I couldn't work it out).

Reply:
This counts the amount of allocatable vCPU you get from summing all the nodes in that zone. So I think this is closest to your second bullet point, I'll work on cleaning this up.

Reply:
https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/ might be relevant then?

Reply:
(that diagram could use work &hellip;)

Reply:
Updated, thanks for the great reference!